### PR TITLE
Add filter box to apps page (fixes #73)

### LINKED ISSUE #73: Add a filter box to the applications page
*This issue is intended as an onboarding task for potential outreachy applicants. Please do not work on it unless you have completed the initial qualification task and it has been assigned to you.*

If you look at the bigquery table view, you'll see that there's a filter box that lets you easily search for the subset of columns that you're interested in:

image

We should have a similar widget on the main page to filter through the list of applications where e.g. putting "firefox" into the box will only show applications with Firefox in the name.

### PR DESCRIPTION
**Pull request by DawoodShahat**
This PR fixes issue #73 Where Search Box is added to the main Application's Page

Screenshot 2020-10-08 at 8 26 07 PM

### CODE REVIEW

File: src/pages/AppList.svelte [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
+  };
 </script>
 
 <h2>Applications</h2>
+<input
+  class="p-2 mb-4"
+  type="text"
+  bind:value={searchTerm}
+  on:input={searchTextChange}
+  placeholder="Search" />
```
Comment (wlach) [inline src/pages/AppList.svelte:26]: The styling we have in the table view component looks a little nicer:

https://github.com/mozilla/glean-dictionary/blob/5e787003843d662079e8c047078c108de8b6d976/src/components/SchemaViewer.svelte#L50

The easiest way to make sure we get this (and that it stays consistent if we ever want to change it) is to refactor the filter box into a seperate component. I'll leave how to do so as an exercise for you to figure out based on the repository layout. 😄 If you need some help, the svelte tutorial is very good: https://svelte.dev/tutorial/basics

**Review by wlach — CHANGES_REQUESTED** [review]
Hi @DawoodShahat! Great start, but let's try to refine the solution a bit. Some suggestions in line.

File: src/components/FilterBox.svelte [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<script>
+  import throttle from "just-throttle";
+
+  export let onChangeText = null;
```
Comment (wlach) [inline src/components/FilterBox.svelte:4]: Pretty sure this doesn't need to be set to null.

```suggestion
  export let onChangeText;
```

File: src/components/SchemaViewer.svelte [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
-  import throttle from "just-throttle";
   import SchemaNode from "./SchemaNode.svelte";
+  import FilterBox from "./FilterBox.svelte";
 
   export let app;
   export let nodes = [];
-  let filterText = "";
   let nodesWithVisibility;
-  const filterTextChanged = () => {
+  const filterTextChanged = (filterText = "") => {
```
Comment (wlach) [inline src/components/SchemaViewer.svelte:8]: I don't think this function needs a default value?

```suggestion
  const filterTextChanged = (filterText) => {
```

File: src/pages/AppList.svelte [hunk 4]
```diff
@@ -1,16 +1,25 @@
 <script>
+  import FilterBox from "../components/FilterBox.svelte";
```
Comment (wlach) [inline src/pages/AppList.svelte:2]: Let's actually call this a FilterInput, as FilterBox is a bit ambiguous (you'll also need to rename the component).

```suggestion
  import FilterInput from "../components/FilterInput.svelte";
```

**Review by wlach — CHANGES_REQUESTED** [review]
This is looking great! Just a few more changes then I think we'll be good to merge.

File: src/components/SchemaViewer.svelte (hunk 3 above)
Comment (DawoodShahat) [reply src/components/SchemaViewer.svelte:8]: This function is called inside SchemaViewer for the initial render. If I remove the default value it throws variable undefined error.

**Comment by wlach** [comment]
Thank you @DawoodShahat! This looks and works great.

**Review by wlach — APPROVED** [review]

**Pull request merged**